test-two/src/services/github.js

```diff
@@ -1,16 +1,64 @@
-import { get } from "../helpers/http";
+import { get, joinPaginatedRequests, post } from "../helpers/http";
 
-const getAuthenticationQueryString = async apiAuthenticationToken =>
-  `?access_token=${apiAuthenticationToken}`;
+const PAGE_SIZE = 100;
+
+const getAuthHeader = apiToken => ({
+  Authorization: `Bearer ${apiToken}`,
+});
 
 export const getUsersForOrganisation = async (
   baseUrl,
+  apiToken,
+  orgId
+) => {
+  const totalUsers =
+    await getTotalUsers(apiToken, orgId);
+
+  return await joinPaginatedRequests(
+    totalUsers,
+    PAGE_SIZE,
+    (pageSize, current) => {
+      return get(
+      `${baseUrl}orgs/${orgId}/members?per_page=${pageSize}&page=${current}`, {
+        headers: {
+          'Content-Type': 'application/json',
+          ...getAuthHeader(apiToken),
+        },
+      }
+    )}
+  )
+};
+
+const getTotalUsers = async (
   apiAuthenticationToken,
   organisationId
 ) => {
-  return get(
-    `${baseUrl}orgs/${organisationId}/members${await getAuthenticationQueryString(
-      apiAuthenticationToken
-    )}`
-  );
-};
+  
+    const response = await post('https://api.github.com/graphql', {
+      headers: {
+        'Content-Type': 'application/json',
+        ...getAuthHeader(apiAuthenticationToken),
+      },
+      body: JSON.stringify({
+        query: `query {
+          organization(login: "${organisationId}") {
+            membersWithRole {
+              totalCount
+            }
+          }
+        }
+      `})
+    });  
+    
+    if(
+      response
+      && response.data
+      && response.data.organization
+      && response.data.organization.membersWithRole
+      && response.data.organization.membersWithRole.totalCount
+    ) {
+      return response.data.organization.membersWithRole.totalCount;
+    }
+
+    throw new Error(`Could not find total of members in ${organisationId}`);
```

Comment:
I kinda prefer the redundant else statement for readability, maybe it's because I'm getting old, what's your preference?

Reply:
I really don't have much of a preference here, I like to have fewer nested lines of code. 
But at the end of the day, there will be optimizations that will render both in the same fashion, to try to improve the execution.
In summary, reading the code is the important point (unless is a heavy hit path, like in the order of thousands or millions per execution).